core/modification/src/main/java/datawave/modification/configuration/ModificationServiceConfiguration.java

```diff
@@ -47,14 +48,18 @@ public void setSecurityMarkingExemptFields(List<String> securityMarkingExemptFie
     /**
      * Handle to query service in case the modification service needs to run queries.
      *
-     * @return RemoteQueryExecutor
+     * @return ModificationQueryService
      */
-    public QueryExecutorBean getQueryService() {
-        return queryService;
+    public ModificationQueryService getQueryService(ProxiedUserDetails userDetails) {
+        return queryServiceFactory.createService(userDetails);
```

Comment:
potential npe

Reply:
throw an IllegalStateException?

Reply:
Both of those are RuntimeExceptions...  If it's all the same to you, I would rather just stick with the NPE.